test/jdk/java/net/httpclient/ShortResponseBody.java

```diff
@@ -503,8 +528,9 @@ public void run() {
                     String query = uriPath.getRawQuery();
                     if (query == null) {
                         out.println("Request headers: [" + headers + "]");
+                        writeResponse(s, BAD_REQUEST_RESPONSE, BAD_REQUEST_RESPONSE.length());
```

Comment:
I'd suggest improving the log message above to say that an unexpected request has been received, and then add a second line to say "Replying with 400 Bad Request".
Otherwise this looks reasonable to me. Sending 400 is a good idea.

Reply:
I pushed [c08924b](https://github.com/openjdk/jdk/pull/9329/commits/c08924bf9e8e8153981f8d75318b510cca873452)